htdocs/core/lib/functions.lib.php

```diff
@@ -1658,8 +1658,10 @@ function dol_syslog($message, $level = LOG_INFO, $ident = 0, $suffixinfilename =
 	if (!empty($message)) {
 		// Test log level
 		$logLevels = array(LOG_EMERG=>'EMERG', LOG_ALERT=>'ALERT', LOG_CRIT=>'CRITICAL', LOG_ERR=>'ERR', LOG_WARNING=>'WARN', LOG_NOTICE=>'NOTICE', LOG_INFO=>'INFO', LOG_DEBUG=>'DEBUG');
+
 		if (!array_key_exists($level, $logLevels)) {
-			throw new Exception('Incorrect log level');
+			dol_syslog($langs->trans('ErrorBadLogLevel', $level), LOG_ERR);
```

Comment:
This is simply a recursive call with forced values to add a valid log entry, indicating that the provided log level is not valid. There is no risk of infinity loop since the parameter used to enter in this condition if forced to a correct value in this call (`LOG_ERR`).

Reply:
The object $langs, $user, $conf may not be always defined.
For exemple when code is called by script, of when constant NOREQUIREDB is defined, or NOREQUIREUSER is defined.